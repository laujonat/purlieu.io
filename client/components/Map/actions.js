export const RECEIVE_CLIENT_LOCATION = "RECEIVE_CLIENT_LOCATION"
export const RECEIVE_CLIENT_LOCATION_SUCCESS = "RECEIVE_CLIENT_LOCATION_SUCCESS"
export const RECEIVE_CLIENT_LOCATION_ERROR = "RECEIVE_CLIENT_LOCATION_ERROR"

export const RECEIVE_MARKER_LOCATION = "RECEIVE_MARKER_LOCATION"
export const RECEIVE_MARKER_LOCATION_SUCCESS = "RECEIVE_MARKER_LOCATION_SUCCESS"
export const RECEIVE_MARKER_LOCATION_ERROR = "RECEIVE_MARKER_LOCATION_ERROR"

// fetching
export const NEW_LOCATION = "NEW_LOCATION"
export const FETCH_LOCATION = "FETCH_LOCATION"

export const fetchLocation = () => ({
  type: FETCH_LOCATION
})

// Map geoLocation
export const receiveClientLocationSuccess = data => ({
  type: RECEIVE_CLIENT_LOCATION_SUCCESS,
  data
})

export const receiveClientLocationErrors = errors => ({
  type: RECEIVE_CLIENT_LOCATION_ERROR,
  errors
})

export const receiveClientLocation = data => ({
  type: RECEIVE_CLIENT_LOCATION,
  data
})

export const receiveMarkerLocationError = errors => ({
  type: RECEIVE_MARKER_LOCATION_ERROR,
  errors
})

// Markers
export const receiveMarkerLocationSuccess = data => ({
  type: RECEIVE_MARKER_LOCATION_SUCCESS,
  data
})

export const receiveMarkerLocation = data => ({
  type: RECEIVE_MARKER_LOCATION,
  data
})
