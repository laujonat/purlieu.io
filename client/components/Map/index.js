export { default } from './map_component';
