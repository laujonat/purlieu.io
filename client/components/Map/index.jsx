import React from "react"
import MapStyle from "./map_style"

export default class Map extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      userLocation: null,
      userAddress: null,
      status: "",
      loading: false
    }

    this.geocoder = new google.maps.Geocoder();
    this.directionsServiceObject = new google.maps.DirectionsService();
    this.directionsRenderer = new google.maps.DirectionsRenderer();
  }

  componentDidMount() {
    this.getUserLocation();
    this.initializeMap();
  }

  mapOptions = center => ({
    center: center,
    zoom: 13,
    zoomControl: false,
    mapTypeControl: false,
    scaleControl: false,
    streetViewControl: false,
    rotateControl: false,
    fullscreenControl: false,
    styles: MapStyle
  })

  initializeMap = () => {
    const sfCenter = {
      lat: 37.773972,
      lng: -122.431297
    };
    const center = this.state.userLocation || sfCenter;

    this.map = new google.maps.Map(
      this.refs.renderedMap,
      this.mapOptions(center)
    );

    this.marker = new google.maps.Marker({
      position: center,
      map: this.map,
      draggable: true
    });

    this.marker.addListener("dragend", () =>
      this.resetMarkerPositionOnClick(this.marker)
    );
    this.marker.addListener("click", () =>
      this.resetMarkerPositionOnClick(this.marker)
    );
    this.map.addListener("click", e => {
      console.log("Marker moved")
      this.marker.setPosition(e.latLng);
      this.resetMarkerPositionOnClick(this.marker);
    });
  }

  geocodeLocation = latLngObject => {
    const geocoder = new google.maps.Geocoder()
    geocoder.geocode({ location: latLngObject }, (results, status) => {
      if (status === "OK") {
        this.setState({ userAddress: results[0].formatted_address })
      }
    })
  }

  resetMarkerPositionOnClick = centerMarker => {
    const newPosition = centerMarker.getPosition()
    this.geocodeLocation(newPosition)
    this.centerMap(newPosition)
  }

  centerMap = locationLatLng => {
    this.map.setCenter(locationLatLng)
    this.marker.setPosition(locationLatLng)
  }

  getUserLocation = () => {
    const successCallback = position => {
      const parsedLocation = {
        lat: position.coords.latitude,
        lng: position.coords.longitude
      }
      this.geocodeLocation(parsedLocation);
      this.marker.setPosition(parsedLocation);
      this.centerMap(parsedLocation);
      this.setState({
        userLocation: parsedLocation,
        status: ''
      });
    };

    const errorCallback = () => {
      this.setState({
        status: "could not fetch location",
        userAddress: null
      });
    };

    this.setState({ status: "fetching location" }, () => {
      navigator.geolocation.getCurrentPosition(successCallback, errorCallback, {
        timeout: 10000,
        enableHighAccuracy: true
      });
    });
  }

  newMarker = pos => {
    new google.maps.Marker({
      position: pos,
      map: this.map,
      title: `${pos.lat()}, ${pos.lng()}`,
      icon: {
        url: "http://maps.google.com/mapfiles/ms/icons/red-dot.png",
        scaledSize: new google.maps.Size(20, 20)
      }
    })
  }

  render() {
    console.log(this.state.status)
    return (
      <div className="map-component">
        <div ref="renderedMap" id="map-container" />
      </div>
    )
  }
}
