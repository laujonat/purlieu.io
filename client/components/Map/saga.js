import { call, put, takeEvery } from "redux-saga/effects"
import {
  RECEIVE_CLIENT_LOCATION,
  RECEIVE_MARKER_LOCATION,
  DRAW_POLYGON,
  receiveClientLocationSuccess,
  receiveClientLocationErrors,
  receiveMarkerLocationSuccess,
  receiveMarkerLocationError,
  receiveDrawPolygonSuccess,
  receiveDrawPolygonError
} from "../Map/actions"

import api from "../../services/map"
import { createPolygon } from "../../lib/map"

export function* fetchClientLocation() {
  try {
    const location = yield call(api.getLocation)
    const address = yield call(api.getAddress, location)
    const data = {
      location,
      address
    }

    yield put(receiveClientLocationSuccess(data))
  } catch (error) {
    yield put(receiveClientLocationErrors(error))
  }
}

export function* setMarkerAddress(geoLocation) {
  const location = geoLocation.data
  try {
    const address = yield call(api.getAddress, location)
    const data = {
      location,
      address
    }
    yield put(receiveMarkerLocationSuccess(data))
  } catch (error) {
    yield put(receiveMarkerLocationError(error))
  }
}

function* drawPolygon({ data }) {
  try {
    const boundaries = yield call(api.getRecalculatedBoundaries, data)
    const polygon = createPolygon(boundaries, data.map)

    yield put(receiveDrawPolygonSuccess({ address: data.address, polygon }))
  } catch (error) {
    yield put(receiveDrawPolygonError(error))
  }
}

export default function*() {
  yield takeEvery(RECEIVE_CLIENT_LOCATION, fetchClientLocation)
  yield takeEvery(RECEIVE_MARKER_LOCATION, setMarkerAddress)
  yield takeEvery(DRAW_POLYGON, drawPolygon)
}
