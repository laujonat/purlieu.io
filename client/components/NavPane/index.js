import React, { Component } from "react"
import styled from "styled-components"
import PropTypes from "prop-types"
import { connect } from "react-redux"
import { spaces } from "../../lib/styles"
import { bindActionCreators } from "redux"

const Container = styled.nav`
  flex: 1 1 auto;
  height: 100vh;
  text-align: center;
  background: #fff;
  box-shadow: 0 0 20px rgba(0, 0, 0, 0.3);
  z-index: 1;
`

const HeaderContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100px;
  align-items: center;
  background-color: #f0f0f5;
  font-weight: 600;
`

const Header = styled.h1``

const DollarInput = styled.input.attrs({ type: "number" })`
  height: 50px;
  background-color: whitesmoke;
  padding: 0 ${spaces.md} 0 ${spaces.md};
`

const AddressInput = styled.input`
  margin-top: ${spaces.md};
  width: 100%;
  height: 50px;
  background-color: whitesmoke;
  padding: 0 ${spaces.md};
`

const SubmitButton = styled.button`
  margin-top: ${spaces.md};
  width: 100%;
  height: 50px;
  background-color: greenyellow;
  padding: 0 ${spaces.md};
`

class NavPane extends Component {
  constructor(props) {
    super(props)

    this.state = {
      dollarInput: ""
    }
  }

  onSubmit = e => {
    e.preventDefault()
  }

  render() {
    return (
      <Container>
        <HeaderContainer>
          <Header>purlieu.io</Header>
        </HeaderContainer>
        <DollarInput />
        <AddressInput />
        <SubmitButton />
      </Container>
    )
  }
}

const mapStateToProps = state => ({
  boundaries: state.entities.lyft.boundaries
})

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      // receiveBoundaries
    },
    dispatch
  )
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NavPane)
