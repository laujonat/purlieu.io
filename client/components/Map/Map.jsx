import React, { Component } from "react"
import { connect } from "react-redux"
import styled from "styled-components"
import PropTypes from "prop-types"
import { mapOptions } from "../../lib/map"
import { media } from "../../lib/styles"
import { receiveMap, receiveClientLocation, receiveMarkerLocation, receiveDrawPolygon } from "./actions"

const Container = styled.div`
  flex: 1 1 70%;

  ${media.tablet`
    height: 50%;
  `}
  ${media.mobileM`
    height: 50%; 
  `}
`

const MapComponent = styled.div`
  height: 100%;
  width: 100%;
`

const google = global.google

class Map extends Component {
  constructor(props) {
    super(props)

    this.renderedMap = React.createRef()
    this.geocoder = new google.maps.Geocoder()
    this.directionsServiceObject = new google.maps.DirectionsService()
    this.directionsRenderer = new google.maps.DirectionsRenderer()
  }

  componentDidMount() {
    this.initializeMap()
    this.getUserLocation()
  }

  componentDidUpdate(prevProps) {
    if (prevProps.location !== this.props.location) {
      this.centerMap(this.props.location)
    }

    if (prevProps.polygonList.length < this.props.polygonList.length) {
      const { drawPolygon, location, address, polygonList } = this.props
      drawPolygon({
        location,
        address,
        boundaries: polygonList[polygonList.length - 1].boundaries,
        map: this.map
      })
    }
  }

  initializeMap = () => {
    const { location, setMap } = this.props
    this.map = new google.maps.Map(this.renderedMap.current, mapOptions(location))
    setMap(this.map)

    this.marker = new google.maps.Marker({
      position: location,
      map: this.map,
      draggable: true
    })

    const markerFunc = e => {
      this.resetMarkerPositionOnClick(this.marker)
      this.setMarkerAddress(e.latLng)
    }

    this.marker.addListener("dragend", markerFunc, false)
    this.marker.addListener("click", markerFunc, false)

    this.map.addListener("click", e => {
      this.marker.setPosition(e.latLng)
      markerFunc(e)
    })
  }

  setMarkerAddress = geoLocation => {
    this.props.setMarkerAddress({
      lat: geoLocation.lat(),
      lng: geoLocation.lng()
    })
  }

  resetMarkerPositionOnClick = centerMarker => {
    const newPosition = centerMarker.getPosition()
    this.centerMap(newPosition)
  }

  centerMap = locationLatLng => {
    this.map.setCenter(locationLatLng)
    this.marker.setPosition(locationLatLng)
  }

  getUserLocation = () => {
    this.props.fetchClientLocation()
  }

  render() {
    return (
      <Container>
        <MapComponent ref={this.renderedMap} />
      </Container>
    )
  }
}

const mapStateToProps = ({ map, polygonList }) => ({
  location: map.location,
  address: map.address,
  polygonList
})

const mapDispatchToProps = dispatch => ({
  setMap: map => dispatch(receiveMap(map)),
  fetchClientLocation: () => dispatch(receiveClientLocation()),
  setMarkerAddress: geoLocation => dispatch(receiveMarkerLocation(geoLocation)),
  drawPolygon: ({ location, address, boundaries, map }) =>
    dispatch(receiveDrawPolygon({ location, address, boundaries, map }))
})

Map.defaultProps = {
  location: {
    lat: 37.773972,
    lng: -122.431297
  },
  address: "",
  polygonList: [],
  setMarkerAddress: () => {},
  drawPolygon: () => {},
  setMap: () => {}
}

Map.propTypes = {
  fetchClientLocation: PropTypes.func,
  setMarkerAddress: PropTypes.func,
  drawPolygon: PropTypes.func,
  setMap: PropTypes.func,
  location: PropTypes.object,
  address: PropTypes.string,
  polygonList: PropTypes.array
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Map)
