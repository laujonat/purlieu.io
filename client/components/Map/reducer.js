import {
  RECEIVE_MAP,
  RECEIVE_CLIENT_LOCATION_SUCCESS,
  RECEIVE_MARKER_LOCATION_SUCCESS,
  DRAW_POLYGON_SUCCESS
} from "./actions"
import { DELETE_POLYGON_CARD } from "../../components/PolygonList/actions"
import { RECEIVE_BOUNDARIES_SUCCESS } from "../../boundaries/actions"

export const initialState = {
  markers: [],
  boundaries: [],
  polygons: []
}

const google = global.google

export const reducer = (state = initialState, action) => {
  Object.freeze(state)
  const newState = { ...state }
  switch (action.type) {
    case RECEIVE_MAP:
      return {
        ...newState,
        map: action.data
      }
    case RECEIVE_CLIENT_LOCATION_SUCCESS:
    case RECEIVE_MARKER_LOCATION_SUCCESS:
      return {
        ...newState,
        ...action.data
      }
    case RECEIVE_BOUNDARIES_SUCCESS: {
      const { marker, boundaries } = action.data

      return {
        ...newState,
        boundaries: [...newState.boundaries, boundaries],
        markers: [...newState.markers, marker]
      }
    }
    case DRAW_POLYGON_SUCCESS: {
      const { polygon } = action.data
      return {
        ...newState,
        polygons: [...newState.polygons, polygon]
      }
    }
    case DELETE_POLYGON_CARD:
      newState.markers[action.data].setMap(null)
      newState.polygons[action.data].setMap(null)
      delete newState.boundaries[action.data]
      newState.markers.splice(action.data, 1)
      newState.polygons.splice(action.data, 1)
      return {
        ...newState,
        markers: newState.markers,
        polygons: newState.polygons
      }
    default:
      return state
  }
}

export default reducer
