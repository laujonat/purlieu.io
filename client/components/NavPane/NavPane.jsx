import React, { Component } from "react"
import styled from "styled-components"
import PropTypes from "prop-types"
import InputRange from 'react-input-range';
import { connect } from "react-redux"
import { spaces } from "../../lib/styles/spaces"
import { receiveBoundaries } from "../../actions"

const Container = styled.nav`
  display: flex;
  flex-direction: column;
  flex: 1 1 auto;
  height: 100vh;
  text-align: center;
  background: #fff;
  box-shadow: 0 0 20px rgba(0, 0, 0, 0.3);
  padding: ${spaces.md};
  min-width: 250px;
`

const HeaderContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100px;
  align-items: center;
  background-color: #f0f0f5;
  font-weight: 600;
`

const Header = styled.h1``

const DollarInputContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`

const Input = styled.input`
  flex: 1;
  margin-top: ${spaces.md};
  padding: 0 ${spaces.md};
  height: 50px;
  width: 100%;
  background-color: whitesmoke;
`

const DollarInput = styled(Input).attrs({
  type: "range",
  min: 10,
  max: 400,
  step: 1
})`
`

const AddressInput = styled(Input)`
  flex: none;
`

const DollarLabel = styled.h1`
  width: 20%;
`


const SubmitButton = styled.button`
  margin-top: ${spaces.md};
  width: 100%;
  height: 50px;
  background-color: greenyellow;
  padding: 0 ${spaces.md};
`

class NavPane extends Component {
  constructor(props) {
    super(props)

    this.state = {
      dollarInput: 10,
      addressInput: ""
    }
  }

  componentDidUpdate(prevProps) {
    if (prevProps.address !== this.props.address) {
      this.setState({ addressInput: this.props.address })
    }
  }

  onSubmit = () => {
    const { dollarInput } = this.state
    const { location } = this.props
    this.props.getBoundaries({
      amount: dollarInput,
      currentLocation: location
    })
  }

  onChange = field => e => {
    this.setState({ [field]: e.target.value })
  }

  render() {
    return (
      <Container>
        <HeaderContainer>
          <Header>purlieu.io</Header>
        </HeaderContainer>
        <DollarInputContainer>
          <DollarInput
            value={this.state.dollarInput}
            onChange={this.onChange("dollarInput")}
          />
          <DollarLabel>{`$${this.state.dollarInput}`}</DollarLabel>
        </DollarInputContainer>
        <AddressInput
          value={this.state.addressInput}
          onChange={this.onChange("addressInput")}
        />
        <SubmitButton onClick={this.onSubmit}>Show Me Dah Wey</SubmitButton>
      </Container>
    )
  }
}

const mapStateToProps = ({ entities: { map } }) => ({
  address: map.clientLocation.address,
  location: map.clientLocation.location
})

const mapDispatchToProps = dispatch => ({
  getBoundaries: ({ amount, currentLocation }) =>
    dispatch(receiveBoundaries({ amount, currentLocation }))
})

NavPane.defaultProps = {
  address: '', 
  location: { },
  getBoundaries: () => { }
}

NavPane.propTypes = {
  getBoundaries: PropTypes.func,
  address: PropTypes.string,
  location: PropTypes.object
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NavPane)
