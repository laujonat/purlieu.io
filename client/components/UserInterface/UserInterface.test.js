import React from "react"
import NavPane from "../NavPane"
import Map from "../Map"
import { shallow } from "enzyme"

describe("UserInterface", () => {
  it("renders the NavPane with no props", () => {
    const NavPaneComponent = shallow(<NavPane />)
    expect(NavPaneComponent).toMatchSnapshot()
  })

  it("renders the NavPane with no props", () => {
    const MapComponent = shallow(<Map />)
    expect(MapComponent).toMatchSnapshot()
  })
})
