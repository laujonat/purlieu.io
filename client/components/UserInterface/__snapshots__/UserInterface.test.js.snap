// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInterface renders the NavPane with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(NavPane) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`UserInterface renders the NavPane with no props 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Map) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
